### ISSUE TITLE
All functions throw errors - cannot connect to IEX API

### ISSUE THREAD

**Issue by benilak** [schardtbc/iexcloudR#1]
The iexcloudR package loads with no problems. My API keys are saved in .env and `Sys.getenv` verifies these variables are saved. My problem is any function I've tried so far returns the same error: 
```
Error in curl::curl_fetch_memory(url, handle = handle) : 
  Could not resolve host: NA
```

The traceback is usually the same except the first (1) function:
```
8: .Call(R_curl_fetch_memory, enc2utf8(url), handle, nonblocking)
7: curl::curl_fetch_memory(url, handle = handle)
6: request_fetch.write_memory(req$output, req$url, handle)
5: request_fetch(req$output, req$url, handle)
4: request_perform(req, hu$handle$handle)
3: httr::GET(url)
2: iex(endpoint)
1: historyFor("AAPL")
```

This is the code for `curl::curl_fetch_memory`:
```
function (url, handle = new_handle()) 
{
  nonblocking <- isTRUE(getOption("curl_interrupt", TRUE))
  output <- .Call(R_curl_fetch_memory, enc2utf8(url), handle, 
    nonblocking)
  res <- handle_data(handle)
  res$content <- output
  res
}
```

Eventually the error traces back to the line `output <- .Call(R_curl_fetch_memory, enc2utf8(url), handle, 
    nonblocking)` in the block above, at which point we've reached C++  and I can't dig any deeper.

**Comment by schardtbc** [comment]
try:
```
res <- iexRaw("/stock/AAPL/quote")
res
## Response [https://cloud.iexapis.com/v1/stock/AAPL/quote?token=pk_your_key]
##  Date: 2019-05-13 23:22
##  Status: 200
##  Content-Type: application/json; charset=utf-8
##  Size: 859 B

```

what response are you getting

**Comment by schardtbc** [comment]
try this as well

res <- httr::GET("https://cloud.iexapis.com/v1/stock/AAPL/quote?token=pk_your_key")

**Comment by benilak** [comment]
`res <- iexRaw("/stock/AAPL/quote")` returns the same error,
however `res <- httr::GET("https://cloud.iexapis.com/v1/stock/AAPL/quote?token=pk_your_key")` passes with the response:
```
Response [https://cloud.iexapis.com/v1/stock/AAPL/quote?token=pk_your_key]
  Date: 2019-05-14 02:13
  Status: 403
  Content-Type: text/html; charset=utf-8
  Size: 34 B
```

**Comment by benilak** [comment]
Additionally, debugging `iexRaw` and running `res <- iexRaw("/stock/AAPL/quote")` shows these values in the process:
endpoint = "/stock/AAPL/quote"
url = "NA/stock/AAPL/quote?token=NA"

which explains why it's throwing the `Could not resolve host: NA` error. This gives me something to work with but I admit I'm not the most proficient coder

**Comment by schardtbc** [comment]
```
readRenviron(".env")
env<-Sys.getenv();
iexcloud <- env[grep("^IEXCLOUD",names(env))];

iexcloud["IEXCLOUD_API_VERSION"]
iexcloud["IEXCLOUD_PUBLIC_KEY"]
```

is the code that gets the environment variables

Does this return what you expect?

**Comment by benilak** [comment]
Yes, I manually went through each step in each of the function's code that led up to `iexRaw` and every variable came out as it should. But for some reason when calling the `iexRaw` alone it would return the error. I ended up reinstalling the package with `keep_sources = TRUE` so I could better debug, but the issue went away after this - the package appears to be working as it should now.

One of my hunches had to with the .env file. I had to create this file myself and put the keys inside; the environment file R creates itself is called .Renviron. Is there a reason your instructions say to put the keys inside this separate .env file that one would have create themselves, and not in the default .Renviron file?

The strangest part out of all of this is that I set IEXCLOUD_API_VERSION = "v1" in both the .env and .Renviron files (it was set to "beta" before, just following the instructions in the README), and yet the response I get from running `iexRaw` shows "beta" in the html. I'm not looking into it further however, since everything seems to be running okay now.
